components/google_drive/sources/new-files/new-files.js

```diff
@@ -0,0 +1,227 @@
+const { uuid } = require("uuidv4");
+const includes = require("lodash.includes");
+const googleDrive = require("../../google_drive.app.js");
+
+module.exports = {
+  key: "google_drive-new-files",
+  name: "New Files (Instant)",
+  description:
+    "Emits a new event any time a new file is added in your linked Google Drive",
+  version: "0.0.1",
+  dedupe: "unique",
+  props: {
+    googleDrive,
+    db: "$.service.db",
+    http: "$.interface.http",
+    drive: { propDefinition: [googleDrive, "watchedDrive"] },
+    folders: {
+      type: "string[]",
+      label: "Folders",
+      description: "The folders you want to watch for changes.",
+      async options({ page, prevContext }) {
+        const { nextPageToken } = prevContext;
+        let results;
+        if (!this.drive) return [];
+        if (this.drive === "myDrive") {
+          results = await this.googleDrive.listFiles({
+            pageToken: nextPageToken,
+            q: "mimeType = 'application/vnd.google-apps.folder'",
+          });
+        } else {
+          results = await this.googleDrive.listFiles({
+            pageToken: nextPageToken,
+            corpora: "drive",
+            driveId: this.drive,
+            includeItemsFromAllDrives: true,
+            supportsAllDrives: true,
+            q: "mimeType = 'application/vnd.google-apps.folder'",
+          });
+        }
+        results.options.unshift({ label: "All Folders", value: "allFolders" });
```

Comment:
I agree. The issues I was running into is that it errors if there are no folders in the user's drive. I suppose we can just assume they'll probably have at least one folder?

Reply:
Not necessarily, a user can in fact have no folders.
We can instead do something like this:
```js
module.exports = {
  props: {
    type: "string[]",
    label: "Folders",
    description: "The folders you want to watch for changes.",
    // Users can leave this prop blank
    optional: true,
    // If the user skips the prop, it will be set to an empty array
    default: [],
    async options({ prevContext }) {
      // Just list the folders, like you did
    },
  },
};
```

Reply:
I tried that originally, as well as initializing the results to [] and appending the list of folders to it. If there are no folders in the drive, it gets errors. This is what I'm seeing:

Configuring prop folders...
panic: please provide options to select from

goroutine 42 [running]:
pd/util.HandleSurveyError(...)
	/home/runner/work/pipedreamin/pipedreamin/cli/util/prompt_select.go:154
pd/util.PromptSelectMulti(0xc0004905d8, 0x7, 0x1a30530, 0x0, 0x0, 0x14f6aa0, 0xc0004722e0, 0x0, 0x0, 0x0)
	/home/runner/work/pipedreamin/pipedreamin/cli/util/prompt_select.go:136 +0x995
pd/core.configurePropRemoteOptions(0x0, 0x0, 0xc0000821c0, 0x40, 0xc000263180, 0x6, 0x6, 0xc000490550, 0x9, 0xc0001f6240, ...)
	/home/runner/work/pipedreamin/pipedreamin/cli/core/configure_props.go:244 +0x8b7
pd/core.configureProp(0x0, 0x0, 0xc0000821c0, 0x40, 0xc000263180, 0x6, 0x6, 0xc000490550, 0x9, 0xc0001f6240, ...)
	/home/runner/work/pipedreamin/pipedreamin/cli/core/configure_props.go:322 +0x3d0
pd/core.ConfigureProps(0x0, 0x0, 0xc0000821c0, 0x40, 0xc000263180, 0x6, 0x6, 0xc000490550, 0x9, 0xc0001f6240, ...)
	/home/runner/work/pipedreamin/pipedreamin/cli/core/configure_props.go:445 +0x1b8
pd/cmd.glob..func6.3.1(0x7ffeefbffc1f, 0x14, 0xc000480180, 0xc000075500, 0xc0004801a0, 0xc0000940e8, 0xc0000940f0, 0x321, 0x15e18b8)
	/home/runner/work/pipedreamin/pipedreamin/cli/cmd/dev.go:156 +0x263
pd/cmd.glob..func6.3()
	/home/runner/work/pipedreamin/pipedreamin/cli/cmd/dev.go:197 +0x72
pd/cmd.glob..func6.4(0xc000138580, 0xc00005ad80, 0xc0000940e0)
	/home/runner/work/pipedreamin/pipedreamin/cli/cmd/dev.go:206 +0x3f
created by pd/cmd.glob..func6
	/home/runner/work/pipedreamin/pipedreamin/cli/cmd/dev.go:205 +0x56a